packages/itwinui-css/src/side-navigation/side-navigation.scss

```diff
@@ -118,8 +121,13 @@ $iui-side-navigation-icon-margins: calc(1.5 * var(--iui-size-m));
       @include iui-icon-style('l');
     }
 
-    &.iui-submenu-open {
-      background-color: var(--iui-color-background-hover);
+    &.iui-submenu-open::after {
+      content: '';
+      position: absolute;
+      right: -2px;
+      width: 2px;
+      height: 100%;
+      background-color: var(--iui-color-background);
```

Comment:
This creates an empty space when hovered. I think we want the background to occupy that space.
![](https://user-images.githubusercontent.com/9084735/200663133-733f1c28-0d3f-4ee6-bdf6-a027504f1451.png)

also, have you thought about how you want it to look when it's active + submenu open? i think we might want to keep the right stripe for consistency.
![](https://user-images.githubusercontent.com/9084735/200663555-5190cfd4-7b71-4fd1-8087-9f6d4f2f52df.png)

Reply:
When hovered, the area where the border once was now matches the button's hover color.
![Screenshot 2022-11-08 at 3 45 28 PM](https://user-images.githubusercontent.com/849817/200671608-affc11b4-1070-4ee7-8442-f5521dcbd06c.png)

When a tab is active AND submenu open, now looks like this:
![Screenshot 2022-11-08 at 3 40 07 PM](https://user-images.githubusercontent.com/849817/200671737-5c245811-312d-4533-b889-c0008333b269.png)

Reply:
Any chance we can get rid of one of those two pseudo elements (`::before` and `::after`)? There is only one stripe so it doesn't make sense. I think we might want to improve the border handling instead.

Reply:
I've removed `::before` and am instead using a linear-gradient to achieve nearly the same visual result.  I did need to change `background-color` to `background` for the active button stripe to make this work.

Reply:
Are you sure you need to change `background-color` to `background` in `button.scss`? I changed it back and it seems to work fine.

Also, transitioning `background` is not recommended for performance reasons, and it also leads to undesirable results in lots of cases.

Reply:
try this:

```css
    &.iui-submenu-open {
      &::before {
        content: '';
        z-index: -1;
        position: absolute;
        inset: 0 -2px 0 0;
        background-color: inherit;
      }
    }
```

Reply:
try this:

```css
    &.iui-submenu-open {
      --_iui-button-active-stripe-color: linear-gradient(
        0deg,
        var(--iui-color-background-accent-muted) 25%,
        var(--iui-color-border-accent) 25%,
        var(--iui-color-border-accent) 75%,
        var(--iui-color-background-accent-muted) 75%
      );

      &::before {
        content: '';
        position: absolute;
        inset: 0 -2px 0 0;
        z-index: -1;
        background-color: inherit;
      }
    }
```

also i think `--_iui-button-active-stripe-color` should be renamed to `--_iui-button-active-stripe-background`.

Reply:
Using your suggested code and renamed CSS variable.  Did have to leave the change in button to use `background`, `background-color` wouldn't accept a `linear-gradient`.